src/PaprController.sol

```diff
@@ -267,13 +267,13 @@ contract PaprController is
         address sendTo,
         ReservoirOracleUnderwriter.OracleInfo calldata oracleInfo
     ) external override {
+        uint256 count = _vaultInfo[auction.nftOwner][auction.auctionAssetContract].count;
         uint256 collateralValueCached = underwritePriceForCollateral(
             auction.auctionAssetContract, ReservoirOracleUnderwriter.PriceKind.TWAP, oracleInfo
-        ) * _vaultInfo[auction.nftOwner][auction.auctionAssetContract].count;
-        bool isLastCollateral = collateralValueCached == 0;
+        ) * count;
 
         uint256 debtCached = _vaultInfo[auction.nftOwner][auction.auctionAssetContract].debt;
-        uint256 maxDebtCached = isLastCollateral ? debtCached : _maxDebt(collateralValueCached, updateTarget());
+        uint256 maxDebtCached = count == 0 ? 0 : _maxDebt(collateralValueCached, updateTarget());
```

Comment:
we do the `count == 0` check twice, is it worth saving that result to a variable? i.e. keeping `isLastCollateral`

Reply:
It would be but we are limited to 16 variables in memory and the gas isn't so bad. Didn't feel like pulling apart.